test/unit/db_tests.js

```diff
@@ -7,11 +7,18 @@ const sinon = require('sinon');
 class MockTopology extends EventEmitter {
   constructor() {
     super();
+    this.s = {
+      promiseLibrary: Promise
```

Comment:
I'm not sure if this is the best way to ensure the `MockTopology` has a `promiseLibrary`, but I figured this was in the right direction.